server/src/main/java/io/seata/server/storage/db/store/LogStoreDataBaseDAO.java

```diff
@@ -351,8 +351,9 @@ public boolean updateBranchTransactionDO(BranchTransactionDO branchTransactionDO
             conn.setAutoCommit(true);
             ps = conn.prepareStatement(sql);
             ps.setInt(1, branchTransactionDO.getStatus());
-            ps.setString(2, branchTransactionDO.getXid());
-            ps.setLong(3, branchTransactionDO.getBranchId());
+            ps.setString(2, branchTransactionDO.getApplicationData());
```

Comment:
move to the last index to prevent compatibility risk.

Reply:
I changed the condition for updated app data.